radixdlt/babylon-wallet-android
# Fix bugs on ImportLegacyWalletViewModel

## Description
* When importing from olympia wallet, only ledger accounts, the app used to crash
* Besides that while I was playing around with it I discovered many cases were we unnecessarily requested for biometrics. Those cases are now fixed. See below for the test cases.

## How to test
* #### Test scenario 1
    * New wallet
    * Import legacy software and hardware accounts. => One biometric + ledger interaction
* #### Test scenario 2 
    * New wallet
    * Import only hardware accounts => Only ledger interaction
 * #### Test scenario 3
    * New wallet
    * Import only software => One biometric
* #### Test scenario 4
   * Do scenario 3
   * Import again with ledger devices and the already included software account => Only ledger interaction
*   #### Test scenario 5
    * Do scenario 1
    * Then do scenario 1 again => **No interaction** with biometrics and ledger
*  #### Test scenario 6
     * New wallet
     * Account Recovery Scan -> Olympia Accounts -> Use Seed Phrase
     * Add the seed phrase used in olympia wallet
     * Proceed (If you happen to import no accounts since they had no activity to the network even better)
     * Then proceed to olympia import
     * Since the seed phrase exists in profile, you will not be asked to provide it again but you will need to provide biometrics **only once**
     * So the result will be => One biometric, + maybe ledger interaction if you include ledger accounts
 *  #### Test scenario 6
    * New wallet
    * Account Recovery Scan -> Olympia Accounts -> Use Seed Phrase
    * Add a random 12 word seed phrase
    * Proceed and complete the recovery of no accounts, just add the seed phrase to the profile.
    * Start olympia import with at least one software account.
    * You will be asked first for a biometrics to check that the saved seed phrase can derive the accounts
    * You will be asked in the end for biometrics again to save the new seed phrase
    * So in general => 2 biometrics, + ledger interaction if you choose to add ledger accounts
